lib/portage/package/ebuild/doebuild.py

```diff
@@ -609,6 +609,7 @@ def doebuild_environment(
             nproc = get_cpu_count()
             if nproc:
                 mysettings["MAKEOPTS"] = "-j%d" % (nproc)
+            mysettings["GNUMAKEFLAGS"] = "--output-sync=line"
```

Comment:
You should check to see if `GNUMAKEFLAGS` is already in mysettings before setting it. Otherwise, you will clobber the user's settings if they have GNUMAKEFLAGS in the environment or in make.conf.